PSKoans/Public/Measure-Karma.ps1

```diff
@@ -83,40 +97,38 @@ function Measure-Karma {
 
                 continue # skip the rest of the function
             }
-
+            
             $KoansPassed = 0
 
-            foreach ($KoanFile in $SortedKoanList.Path) {
-                Write-Verbose "Testing karma with file [$KoanFile]"
+            foreach ($Koan in $SortedKoanList) {
                 $PesterParams = @{
-                    Script   = $KoanFile
                     PassThru = $true
                     Show     = 'None'
                 }
-                $PesterTests = Invoke-Pester @PesterParams
+                $PesterTests = $Koan.InvokePester($PesterParams)
                 $KoansPassed += $PesterTests.PassedCount
-
+                
                 Write-Verbose "Karma: $KoansPassed"
-                if ($PesterTests.FailedCount -gt 0) {
+                if ($Koan.GetFailedCount() -gt 0) {
                     Write-Verbose "Your karma has been damaged."
                     break
                 }
             }
 
-            if ($PesterTests.FailedCount -gt 0) {
-                $NextKoanFailed = $PesterTests.TestResult |
-                    Where-Object Result -eq 'Failed' |
-                    Select-Object -First 1
-
+            $KoanFailed = $SortedKoanList.GetTestResults('Failed') | Select-Object -First 1
+            if ($KoanFailed) {
                 $Meditation = @{
-                    DescribeName = $NextKoanFailed.Describe
-                    Expectation  = $NextKoanFailed.ErrorRecord
-                    ItName       = $NextKoanFailed.Name
-                    Meditation   = $NextKoanFailed.StackTrace
+                    DescribeName = $KoanFailed.Describe
+                    Expectation  = $KoanFailed.ErrorRecord
+                    ItName       = $KoanFailed.Name
+                    Meditation   = $KoanFailed.StackTrace
                     KoansPassed  = $KoansPassed
                     TotalKoans   = $TotalKoans
                 }
                 Write-MeditationPrompt @Meditation
+
+                #Invoke Koan Help Data.
+                $SortedKoanList | Where-Object { $_.GetFailedCount() -gt 0 } | Select-Object -First 1 | ForEach-Object InvokeHelpInfo
```

Comment:
This should probably be split over at least two, possibly more, lines. Getting a bit lengthy here. 😄